subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/Kubernetes.kt

```diff
@@ -59,3 +65,25 @@ private fun oauthTokenProvider(config: File): OAuthTokenProvider {
 
     return OAuthTokenProvider { kubeConfig.accessToken }
 }
+
+val Project.kubernetesCredentials: KubernetesCredentials
+    get() {
+        val context = getOptionalStringProperty("kubernetesContext", nullIfBlank = true)
```

Comment:
We shouldn't branch login by one specific parameter from authentication parameters.

Service auth = `kubernetesToken` + `kubernetesCaCertData` + `kubernetesUrl`
Open id auth = `kubernetesContext` + `kubernetesCaCertFile`

If I don't have `kubernetesContext`, it doesn't imply that I want to use service account.

Reply:
My proposal is to declare auth credential by extension explicitly. 
A user has to tell what to use and pass required parameters.
Not in this PR

Reply:
Discussed by voice. We decided to implement this logic like a part of emulator target refactoring